src/components/Button/Button.tsx

```diff
@@ -0,0 +1,72 @@
+import { twMerge } from 'tailwind-merge';
+import { ComponentPropsWithoutRef } from 'react';
+
+interface ButtonProps extends ComponentPropsWithoutRef<'button'> {
+  variant?: 'filled' | 'outlined' | 'tonal';
+  shape?: 'rectangle' | 'rounded';
+  size?: 'small' | 'default' | 'large';
+}
+
+export default function Button({
+  className,
+  children,
+  type,
+  variant = 'filled',
+  shape = 'rounded',
+  size = 'default',
+  disabled = false,
+  ...rest
+}: ButtonProps) {
+  const baseStyle = 'flex items-center justify-center outline-none';
+
+  const variantEnabledStyles = {
+    filled: 'bg-primary-500 text-white',
+    outlined: 'text-primary-500 border-primary-200 border-[0.1rem]',
+    tonal: 'bg-primary-50 text-primary-500',
+  };
+
+  const variantDisabledStyles = {
+    filled: 'bg-gray-light text-[#969696]',
+    outlined: 'bg-gray-light text-[#969696] border-[#c3c3c3] border-[0.1rem]',
+    tonal: 'bg-purple-light text-primary-100',
+  };
+
+  const shapeStyles = {
+    rounded: 'rounded-full',
+    rectangle: 'rounded-[0.8rem]',
+  };
+
+  const sizeStyles = {
+    small: 'w-[10.1rem] h-[4rem] px-[2rem] py-[0.8rem] text-[2rem] font-[300]',
+    default: 'w-[12.1rem] h-[5.6rem] px-[2.4rem] py-[1.6rem] text-[2.4rem] font-[300]',
+    large: 'w-[19.6rem] h-[5.6rem] px-[2rem] py-[1.6rem] text-[2.4rem] font-bold',
+  };
+
+  const interactionStyles = {
+    filled: 'hover:shadow-button',
+    outlined:
+      'hover:bg-[rgba(97,0,255,0.08)] hover:shadow-button focus:bg-[rgba(97,0,255,0.12)] focus:border-primary-500 active:bg-[rgba(97, 0, 255, 0.12)] active:border-primary-200',
+    tonal:
+      'hover:bg-[rgba(0,0,0,0.08)] hover:shadow-button focus:bg-[rgba(0,0,0,0.12)] active:bg-[rgba(0,0,0,0.12)]',
+  };
+
+  const combinedClasses = twMerge(
+    baseStyle,
+    shapeStyles[shape],
+    sizeStyles[size],
+    disabled ? variantDisabledStyles[variant] : variantEnabledStyles[variant],
+    !disabled && interactionStyles[variant],
+    className,
+  );
+
+  return (
+    <button
+      className={combinedClasses}
+      type={type === 'submit' ? 'submit' : 'button'}
```

Comment:
P5 아마 린트가 꼭 타입 쓰라 해서 이렇게 하신거 같은데!
```suggestion
      type={type}
```
이렇게 하는게 조금 나을 거 같아요 버튼타입이 하나 더 있긴 해성,,
[type](https://developer.mozilla.org/ko/docs/Web/HTML/Element/button#type)
```
버튼의 행동 방식. 가능한 값은 다음과 같습니다.

submit: 버튼이 서버로 양식 데이터를 제출합니다. 지정하지 않은 경우 기본값이며, 유효하지 않은 값일 때도 사용합니다.
reset: <input type="reset">처럼, 모든 컨트롤을 초깃값으로 되돌립니다.
button: 기본 행동이 없으며 클릭했을 때 아무것도 하지 않습니다. 클라이언트측 스크립트와 연결할 수 있습니다.
```

Reply:
처음에 말씀해주신 것처럼 type={type} 이렇게 썼었는데 린트가 이것도 오류로 잡더라구요...
[Disallow usage of button elements without an explicit type attribute (react/button-has-type)](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/button-has-type.md
)
`var Hello = <button type={condition ? "button" : "submit"}>Hello</button>`
이게 그래도 올바른 코드라고 하는데 이렇게 수정하는 건 어떨까용?!